pkg/specgen/generate/validate.go

```diff
@@ -11,21 +11,21 @@ import (
 	"github.com/containers/podman/v4/pkg/rootless"
 	"github.com/containers/podman/v4/pkg/specgen"
 	"github.com/containers/podman/v4/utils"
+	spec "github.com/opencontainers/runtime-spec/specs-go"
 )
 
 // Verify resource limits are sanely set when running on cgroup v1.
 func verifyContainerResourcesCgroupV1(s *specgen.SpecGenerator) ([]string, error) {
+	var resourceNil *spec.LinuxResources
```

Comment:
Is this variable still needed?  Does the `!= nil` check not work?

Reply:
`s.ResourceLimits` is not nil when the container created by `podman kube play`. 

<details>
<summary>s.ResourceLimits</summary>

```
(dlv) p s.ResourceLimits
*github.com/opencontainers/runtime-spec/specs-go.LinuxResources {
        Devices: []github.com/opencontainers/runtime-spec/specs-go.LinuxDeviceCgroup len: 0, cap: 0, nil,
        Memory: *github.com/opencontainers/runtime-spec/specs-go.LinuxMemory nil,
        CPU: *github.com/opencontainers/runtime-spec/specs-go.LinuxCPU nil,
        Pids: *github.com/opencontainers/runtime-spec/specs-go.LinuxPids nil,
        BlockIO: *github.com/opencontainers/runtime-spec/specs-go.LinuxBlockIO nil,
        HugepageLimits: []github.com/opencontainers/runtime-spec/specs-go.LinuxHugepageLimit len: 0, cap: 0, nil,
        Network: *github.com/opencontainers/runtime-spec/specs-go.LinuxNetwork nil,
        Rdma: map[string]github.com/opencontainers/runtime-spec/specs-go.LinuxRdma nil,
        Unified: map[string]string nil,}
```

</details>

I think `s.ResourceLimits` is equal to `var resourceNil *spec.LinuxResources`.